src/v/kafka/requests/fetch_request.cc

```diff
@@ -458,13 +458,6 @@ static ss::future<> fetch_topic_partition(
     auto& topic = *p.topic;
     auto& part = *p.partition;
 
-    if (
```

Comment:
i can't seem to square the commit message with this delta. it seems like we are removing the block of code that is intended to handle the condition described by the commit message...

Reply:
is this relying on the timeout of the log reader to return an empty response? if so, please try to error on the side of explaining second order effects like this in the commit message. the best case scenario for someone reading this in the future is that its merely a guess that that is what is happening.